### DISCUSSION TITLE
Options to emphasize compactness and increased packing of designed regions in the context of the overall structure?

### DISCUSSION THREAD

**Discussion by tsamdin** [RosettaCommons/rosetta#120] [Rosetta]
Are there any task operations (or other options) to encourage compactness or increased packing? I'm able to use Rosetta to redesign a beta-hairpin in a protein of interest with modifications my group would like to explore at the bench. But the redesigned beta-hairpin ends up "pushed away" from the rest of the protein, breaking contacts with other beta-strands that I'd like to maintain. I'm playing with atom pair constraints at the moment, but was wondering if there were other approaches I could/should use.

**Comment by tsamdin** [comment]
The AtomPairConstraintGenerator [https://docs.rosettacommons.org/docs/latest/scripting_documentation/RosettaScripts/Movers/constraint_generators/AtomPairConstraintGenerator](url) has an example where secondary structure selectors are used to generate constrains. In my case, this seemed to do the trick of keeping the specified motif's closer together. I saw an increase in the number of output structures that were more compact and less diffuse. I also added a turned on the voids_penalty score term as well.

**Comment by NooraAz** [comment]
Maybe using the radius of gyration as a filter.
